src/ansi-c/c_typecheck_base.h

```diff
@@ -205,6 +205,7 @@ class c_typecheck_baset:
     const symbol_exprt &function_symbol);
   virtual exprt
   typecheck_shuffle_vector(const side_effect_expr_function_callt &expr);
+  virtual void disallow_history_variables(exprt &expr);
```

Comment:
This needs to be
`void disallow_history_variables(const exprt &) const;`
No need for this to be virtual, the parameter must be a const-ref, and the method itself is const.